addons/website_sale/controllers/reorder.py

```diff
@@ -4,78 +4,67 @@
+        return: The number of items in the cart.
+        rtype: int
+        """
         try:
             sale_order = self._document_check_access('sale.order', order_id, access_token=access_token)
         except (AccessError, MissingError):
             return request.redirect('/my')
 
-        currency = request.env['website'].get_current_website().currency_id
-        result = {
-            'currency': currency.id,
-            'products': [],
-        }
+        Cart_controller = Cart()
         for line in sale_order.order_line:
-            if not line._show_in_cart():
-                continue
+            if not line.with_user(request.env.user).sudo()._is_reorder_allowed():
+                continue  # Skip section headers, deliveries
 
-            selected_combo_items = []
+            linked_products = []
             if line.product_id.type == 'combo':
                 for linked_line in line.linked_line_ids.filtered('combo_item_id'):
-                    selected_combo_items.append({
-                        **self._get_common_order_line_data(linked_line),
+                    combination = (
+                        linked_line.product_id.product_template_attribute_value_ids
+                        | linked_line.product_no_variant_attribute_value_ids
+                    )
+                    linked_products.append({
+                        'product_template_id': linked_line.product_id.product_tmpl_id.id,
+                        'product_id': linked_line.product_id.id,
+                        'combination': combination.ids,
+                        'no_variant_attribute_value_ids': linked_line.product_no_variant_attribute_value_ids.ids,
+                        'product_custom_attribute_values': [{
+                            'custom_product_template_attribute_value_id': pcav.custom_product_template_attribute_value_id.id,
+                            'custom_value': pcav.custom_value,
+                        } for pcav in linked_line.product_custom_attribute_value_ids],
+                        'quantity': linked_line.product_uom_qty,
                         'combo_item_id': linked_line.combo_item_id.id,
                         'parent_product_template_id': line.product_id.product_tmpl_id.id,
                     })
 
-            add_to_cart_allowed = line.with_user(request.env.user).sudo()._is_reorder_allowed()
-            res = {
-                **self._get_common_order_line_data(line, add_to_cart_allowed),
-                'type': line.product_id.type,
-                'name': line.name_short,
-                'description_sale': line.product_id.description_sale or '' + line._get_sale_order_line_multiline_description_variants(),
-                'add_to_cart_allowed': add_to_cart_allowed,
-                'has_image': bool(line.product_id.image_128),
-                'selected_combo_items': selected_combo_items,
-            }
+            Cart_controller.add_to_cart(
```

Comment:
I believe it's not a good idea to disregard the feedback values given by the add_to_cart call.
You do handle the stock errors, but what about other errors returned by the call?
Also, what about Google Analytics tracking? You don't handle the `tracking_info` content that is returned by the call and should be handled client-side.

I doubt it was handled in the previous implementation of the reordering wizard, but I feel like it's strange to disregard the standard behavior of cart updates when triggered by the reordering logic.